### PR TITLE
Fix for route:cache problem with nested route files

### PR DESCRIPTION
**Pull request by rogue780**
This PR relates to Issue [7522](https://github.com/laravel/framework/issues/7522)

### CODE REVIEW

File: src/Illuminate/Foundation/Console/RouteCacheCommand.php [hunk 1]
```diff
@@ -30,13 +45,15 @@ class RouteCacheCommand extends Command {
...
+	 * @param  \Illuminate\Filesystem\Filesystem $files
+	 * @param  \Illuminate\Routing\Router  $router
 	 */
-	public function __construct(Filesystem $files)
+	public function __construct(Filesystem $files, Router $router)
 	{
 		parent::__construct();
 
+		$this->router = $router;
+		$this->routes = $router->getRoutes();
```
Comment (crynobone) [inline src/Illuminate/Foundation/Console/RouteCacheCommand.php]: If you are loading from current request, which is before running `route:clear`, you simply loading available routes from cached, this is why @taylorotwell load routes from a fresh application.

Comment (rogue780) [inline src/Illuminate/Foundation/Console/RouteCacheCommand.php]: Unfortunately that isn't working right. After i wake up I'll push an
example app to github that demonstrates the problem. Hopefully we can work
together to find a solution.

Comment (crynobone) [inline src/Illuminate/Foundation/Console/RouteCacheCommand.php]: > Unfortunately that isn't working right.

Well, this break everyone else code that using `php artisan route:cache` including myself.

**Comment by GrahamCampbell** [comment]
@crynobone is correct.

**Comment by rogue780** [comment]
Why close the issue? The issue is not resolved, merely an example fix
didn't solve it.

**Comment by rogue780** [comment]
This was just an example that I uploaded as suggested by another user. was not intended to be merged in.

**Pull request closed without merging**